plugins/verifier/vulnerabilityreport/vulnerability_report.go

```diff
@@ -0,0 +1,400 @@
+*/
+
+package main
+
+import (
+	"context"
+	"embed"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	_ "github.com/deislabs/ratify/pkg/referrerstore/oras"
+	"github.com/deislabs/ratify/pkg/verifier"
+	"github.com/deislabs/ratify/pkg/verifier/plugin/skel"
+	"github.com/deislabs/ratify/plugins/verifier/vulnerabilityreport/schemavalidation"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+)
+
+//go:embed schemavalidation/schemas
+var embeddedFS embed.FS
+
+const (
+	SarifMediaType       string = "application/sarif+json"
+	SarifOfflineFilePath string = "schemavalidation/schemas/sarif-2.1.0.json"
+	TrivyScannerName     string = "trivy"
+	GrypeScannerName     string = "grype"
+	CreatedAnnotation    string = imagespec.AnnotationCreated
+	SeverityRegex               = `Severity:\s*(\w+)`
+)
+
+type PluginConfig struct {
+	Name               string   `json:"name"`
+	SchemaURL          string   `json:"schemaURL,omitempty"`
+	MaximumAge         string   `json:"maximumAge,omitempty"`
+	DisallowedSeverity []string `json:"disallowedSeverity,omitempty"`
+	Passthrough        bool     `json:"passthrough,omitempty"`
+	DenylistCVEs       []string `json:"denylistCVEs,omitempty"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("vulnerabilityreport", "1.0.0", VerifyReference, []string{"1.0.0"})
+}
+
+func parseInput(stdin []byte) (*PluginConfig, error) {
+	conf := PluginInputConfig{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse stdin for the input: %w", err)
+	}
+
+	return &conf.Config, nil
+}
+
+func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, referenceDescriptor ocispecs.ReferenceDescriptor, referrerStore referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	input, err := parseInput(args.StdinData)
+	if err != nil {
+		return nil, err
+	}
+
+	// check report is newer than allowed maximum age
+	if input.MaximumAge != "" {
+		ok, created, err := validateMaximumAge(input.MaximumAge, referenceDescriptor)
+		if err != nil {
+			return &verifier.VerifierResult{
+				Name:      input.Name,
+				IsSuccess: false,
+				Message:   fmt.Sprintf("vulnerability report validation failed: error validating maximum age:[%v]", err.Error()),
+			}, nil
+		}
+		if !ok {
+			return &verifier.VerifierResult{
+				Name:      input.Name,
+				IsSuccess: false,
+				Message:   fmt.Sprintf("vulnerability report validation failed: report is older than maximum age:[%s]", input.MaximumAge),
+				Extensions: map[string]interface{}{
+					CreatedAnnotation: created,
+				},
+			}, nil
+		}
+	}
+
+	ctx := context.Background()
+
+	referenceManifest, err := referrerStore.GetReferenceManifest(ctx, subjectReference, referenceDescriptor)
```

Comment:
seems we're adding more coupling between store and verifier though we need it for now.

Reply:
yes that's true. This seems to be the same pattern adopted by other verifiers as well.